go/src/koding/klient/machine/index/node/diagnose.go

```diff
@@ -0,0 +1,260 @@
+package node
+
+import (
+	"fmt"
+)
+
+// Diagnose checks tree looking for broken filesystem invariants. It returns
+// a list of found problems. Each of them should be considered critical since
+// they indicate broken logic.
+func (t *Tree) Diagnose() (s []string) {
```

Comment:
:)

[wiki](https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters) discourages that 

> Don't name result parameters just to avoid declaring a var inside the function; that trades off a minor implementation brevity at the cost of unnecessary API verbosity.